File: kotlin-node/karakum/src/jsMain/resources/plugins/convertStatSyncFnHeritageClause.js
import ts from "typescript";

export default function (node, context, render) {
    if (
        ts.isHeritageClause(node)

        && node.parent
        && ts.isInterfaceDeclaration(node.parent)
        && node.parent.name.text === "StatSyncFn"
    ) {
        return ""
    }

    return null
}

File: kotlin-node/karakum/src/jsMain/kotlin/node/karakum/plugins/convertStatSyncFnHeritageClause.kt
package node.karakum.plugins

import node.karakum.util.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import io.github.sgrishchenko.karakum.util.getParentOrNull
import typescript.isHeritageClause
import typescript.isInterfaceDeclaration

val convertStatSyncFnHeritageClause = createPlugin { node, _, _ ->
    nullable {
        ensure(isHeritageClause(node))

        val interfaceNode = ensureNotNull(node.getParentOrNull())
        ensure(isInterfaceDeclaration(interfaceNode))
        ensure(interfaceNode.name.text == "StatSyncFn")

        ""
    }
}
